src/JsonSchema/SchemaFactory.php

```diff
@@ -194,8 +201,18 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
             }
 
             $subSchema = $this->buildSchema($className, $format, Schema::TYPE_OUTPUT, null, $subSchema, $serializerContext + [self::FORCE_SUBSCHEMA => true], false);
-            $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
-            // prevent "type" and "anyOf" conflict
+            if ($isCollection) {
+                $propertySchema['items']['$ref'] = $subSchema['$ref'];
+                unset($propertySchema['items']['type']);
+                break;
+            }
+
+            if ($type->isNullable()) {
+                $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
```

Comment:
IMHO `oneOf` should be more persistent in this case: https://json-schema.org/understanding-json-schema/reference/combining.html#id7
```suggestion
                $propertySchema['oneOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
```

Reply:
you can probably have null values in a collection with objects?

Reply:
Like `['/books/{id}, null]`? Not sure this is a valid value for us, and is what we want...